test/index.spec.js

```diff
@@ -228,25 +229,193 @@ describe('FlipMove', () => {
     });
   });
 
-  describe('duration prop runtime checking', () => {
-    let wrapper;
-    beforeEach(() => {
-      wrapper = shallow(<FlipMove />);
+  describe('prop runtime checking and conversion', () => {
+    let errorStub;
+    let warnStub;
+    const SFC = () => null;
+
+    before(() => {
+      errorStub = sinon.stub(console, 'error');
+      warnStub = sinon.stub(console, 'warn');
+    });
+    afterEach(() => {
+      errorStub.reset();
+      warnStub.reset();
     });
+    after(() => {
+      errorStub.restore();
+      warnStub.restore();
+    });
+
+    it('doesn\'t run checks in production environment', () => {
+      const env = process.env;
+      process.env = { NODE_ENV: 'production' };
 
-    it('applies a bogus string', () => {
-      wrapper.setProps({ duration: 'hi' });
-      expect(consoleStub).to.have.been.calledOnce;
+      shallow(<FlipMove><SFC key="hi" /></FlipMove>);
+      shallow(<FlipMove>hi</FlipMove>);
+      shallow(<FlipMove disableAnimations />);
+      shallow(<FlipMove duration="hi" />);
+      shallow(<FlipMove appearAnimation="unknown" />);
+
+      expect(errorStub).to.not.have.been.called;
+      expect(warnStub).to.not.have.been.called;
+      process.env = env;
     });
 
-    it('applies an array prop and throws', () => {
-      wrapper.setProps({ duration: ['hi'] });
-      expect(consoleStub).to.have.been.called;
+    describe('timing props', () => {
+      it('applies a bogus string', () => {
+        shallow(<FlipMove duration="hi" />);
+        expect(errorStub).to.have.been.calledWith(`
+>> Error, via react-flip-move <<
```

Comment:
Maybe we should import the error string constructors and use them? That way we don't have to worry about updating the messages in both places if/when they change.

Could also be smart to add simple tests for the error message factories themselves... although I'm not too worried about it, since it's so simple, just string interpolation.

Reply:
By not importing those but writing out the expected output, we kinda test the factories as well